### ISSUE TITLE
Feature request: Add --skip-empty to DOLT_COMMIT()

### ISSUE THREAD

**Issue by jacobATcertifiedfoodDotinfo** [dolthub/dolt#5678]
https://docs.dolthub.com/sql-reference/version-control/dolt-sql-procedures#dolt_commit

Please add an option similar to

> --allow-empty: Allow recording a commit that has the exact same data as its sole parent. This is usually a mistake, so it is disabled by default. This option bypasses that safety.

The new `--skip-empty` option covers a different use case, because I want to commit ONLY if there is something to commit. I do NOT want to get an error like `Error 1105 (HY000): nothing to commit`.

**Comment by timsehn** [comment]
There's no Git equivalent of this. 

https://git-scm.com/docs/git-commit

I assume you want this for the programmatic use case. You don't want to check that something is changed in the working set before issuing the commit command.

**Comment by jacobATcertifiedfoodDotinfo** [comment]
Yes, correct.
Currently, here is a snippet:

```
		_, err = txClient.ExecContext(ctx, commitCmd)
		if err != nil {
			if !strings.Contains(err.Error(), "nothing to commit") {
				return nil, err
			}
		}
```

I don't want to have to check for this error

**Comment by fulghum** [comment]
Hey @jacobATcertifiedfoodDotinfo, thank you for this great feature request! 🙏  I've just merged in PR https://github.com/dolthub/dolt/pull/6086, which adds support for the new `--skip-empty` flag to `dolt commit` on the CLI and `dolt_commit()` in SQL. This will go out in the next release, later this week. 

Let us know if you have any more feedback on anything we can do to make Dolt work better for you!

**Comment by fulghum** [comment]
And... it's released now! You can find it in [Dolt version 1.3.0](https://github.com/dolthub/dolt/releases/tag/v1.3.0) or higher.
